golangci.yaml

```diff
@@ -1,12 +1,35 @@
-linters:
-  disable:
-    - megacheck
+liners:
+  disable-all: true
   enable:
+    - bodyclose
+    - deadcode
+    - depguard
+    - dogsled
+    - errcheck
+    - gocyclo
+    - gofmt
+    - goimports
+    - goprintffuncname
+    - gosec
     - gosimple
+    - govet
+    - ineffassign
+    - interfacer
+    - nakedret
+    - nolintlint
+    - rowserrcheck
+    - scopelint
     - staticcheck
+    - structcheck
+    - typecheck
+    - unconvert
+    - varcheck
+    - gocritic
```

Comment:
`gocritic ` is highly opinionated, but let's see if we get used to it.

Reply:
That one was interesting - I was fully expecting to eliminate it for noise, but actually out our code was already almost 100% compliant! It seems like we share its opinions :) We can remove if it turns out to be a pain.